src/JustEat.StatsD/TimerExtensions.cs

```diff
@@ -120,6 +138,9 @@ public static T Time<T>(this IStatsDPublisher publisher, string bucket, Func<T>
         /// <returns>
         /// The value from invoking <paramref name="func"/>.
         /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="publisher"/> or <paramref name="bucket"/> is <see langword="null"/>.
+        /// </exception>
```

Comment:
Should we do anything special if the `func` is null for these cases?

Reply:
I was pondering that - either we do nothing (as we did before) and let people trip over their `NullReferenceException`s, or I add null-checking to all these methods.

Reply:
You wouldn't really document `NullReferenceException` because that's just a "bug". Either we add the null-checking and document, or leave as-is.

I'd lean towards to the null-check even if it's adding an extra CPU instruction to each method.

Reply:
Have added null checks in f8a61978d2b122d66b925918bab32d99d14e0f3e.